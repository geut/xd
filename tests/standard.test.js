const runCommonTests = require('.')

runCommonTests('standard')
