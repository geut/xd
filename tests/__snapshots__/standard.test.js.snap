// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`standard: fix file 1`] = `"[{\\"filePath\\":\\"/tmp/xd-standard-test/index.js\\",\\"messages\\":[],\\"errorCount\\":0,\\"warningCount\\":0,\\"fixableErrorCount\\":0,\\"fixableWarningCount\\":0,\\"output\\":\\"function test () {\\\\n  console.log('test')\\\\n}\\\\n\\\\ntest()\\\\n\\"}]"`;

exports[`standard: fix file 2`] = `"[{\\"filePath\\":\\"<text>\\",\\"messages\\":[],\\"errorCount\\":0,\\"warningCount\\":0,\\"fixableErrorCount\\":0,\\"fixableWarningCount\\":0,\\"output\\":\\"function test () {\\\\n  console.log('test')\\\\n}\\\\n\\\\ntest()\\\\n\\"}]"`;

exports[`standard: fix file 3`] = `"[{\\"filePath\\":\\"/tmp/xd-standard-test/index.js\\",\\"messages\\":[],\\"errorCount\\":0,\\"warningCount\\":0,\\"fixableErrorCount\\":0,\\"fixableWarningCount\\":0,\\"output\\":\\"function test () {\\\\n  console.log('test')\\\\n}\\\\n\\\\ntest()\\\\n\\"}]"`;

exports[`standard: fix file 4`] = `"[{\\"filePath\\":\\"/tmp/xd-standard-test/index.js\\",\\"messages\\":[],\\"errorCount\\":0,\\"warningCount\\":0,\\"fixableErrorCount\\":0,\\"fixableWarningCount\\":0,\\"output\\":\\"function test () {\\\\n  console.log('test')\\\\n}\\\\n\\\\ntest()\\\\n\\"}]"`;

exports[`standard: lint file 1`] = `
"[0m[0m
[0m[4m/tmp/xd-standard-test/index.js[24m[0m
[0m  [2m1:14[22m  [31merror[39m  Missing space before function parentheses  [2mspace-before-function-paren[22m[0m
[0m  [2m2:15[22m  [31merror[39m  Strings must use singlequote               [2mquotes[22m[0m
[0m  [2m2:22[22m  [31merror[39m  Extra semicolon                            [2msemi[22m[0m
[0m  [2m5:7[22m   [31merror[39m  Extra semicolon                            [2msemi[22m[0m
[0m[0m
[0m[31m[1mâœ– 4 problems (4 errors, 0 warnings)[22m[39m[0m
[0m[31m[1m[22m[39m[31m[1m  4 errors and 0 warnings potentially fixable with the \`--fix\` option.[22m[39m[0m
[0m[31m[1m[22m[39m[0m
# exit 1"
`;

exports[`standard: lint file 2`] = `
"[{\\"filePath\\":\\"/tmp/xd-standard-test/index.js\\",\\"messages\\":[{\\"ruleId\\":\\"space-before-function-paren\\",\\"severity\\":2,\\"message\\":\\"Missing space before function parentheses.\\",\\"line\\":1,\\"column\\":14,\\"nodeType\\":\\"FunctionDeclaration\\",\\"fix\\":{\\"range\\":[13,13],\\"text\\":\\" \\"}},{\\"ruleId\\":\\"quotes\\",\\"severity\\":2,\\"message\\":\\"Strings must use singlequote.\\",\\"line\\":2,\\"column\\":15,\\"nodeType\\":\\"Literal\\",\\"endLine\\":2,\\"endColumn\\":21,\\"fix\\":{\\"range\\":[32,38],\\"text\\":\\"'test'\\"}},{\\"ruleId\\":\\"semi\\",\\"severity\\":2,\\"message\\":\\"Extra semicolon.\\",\\"line\\":2,\\"column\\":22,\\"nodeType\\":\\"ExpressionStatement\\",\\"fix\\":{\\"range\\":[38,42],\\"text\\":\\")\\\\n}\\"}},{\\"ruleId\\":\\"semi\\",\\"severity\\":2,\\"message\\":\\"Extra semicolon.\\",\\"line\\":5,\\"column\\":7,\\"nodeType\\":\\"ExpressionStatement\\",\\"fix\\":{\\"range\\":[49,51],\\"text\\":\\")\\"}}],\\"errorCount\\":4,\\"warningCount\\":0,\\"fixableErrorCount\\":4,\\"fixableWarningCount\\":0,\\"source\\":\\"function test() {\\\\n  console.log(\\\\\\"test\\\\\\");\\\\n}\\\\n\\\\ntest();\\\\n\\"}]
# exit 1"
`;
